### ISSUE TITLE
Merge error - duplicate sample names

### ISSUE THREAD

**Issue by desmodus1984** [samtools/bcftools#2209]
Hi,

I am trying to merge vcf samples obtained from bisulfite data.
I used cgmaptools to call variants from sorted-bam files and when I tried to merge two files as a test, I got the error:
Error: Duplicate sample names (NA00001), use --force-samples to proceed anyway.

Do you know a way to (re)name the files?
I used stats to see the merged file and I didn't get any sample names.
My concern is that I have 44 files and I would like to id the samples with their variants because I need to do assess population differentiation.

Or, is there a way to rename the samples in the multi-sample vcf file?

Thanks;

Juan Pablo

**Comment by pd3** [comment]
`bcftools reheader -s` allows to rename samples. As suggested by the error message, `bcftools merge --force-samples` can be used to merge the files despite duplicate sample names, renaming as described in the manual page http://samtools.github.io/bcftools/bcftools.html#merge

I hope this helps

**Comment by desmodus1984** [comment]
Hi,
I was trying to use a wild card to merge files
Bcftools merge *.bsg.vcf.gz -Oz -o test.vcf.gz
And I got the same error.

do I have to rename each file individually before merging
To
The other question, when using the wild card, how can I tell the order of the file names during merging?

All my samples have the same names at the end of the header.
#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT **V02055.bsg**
 I don't understand why the error now.
Each header of each file has its own file name

**Comment by pd3** [comment]
First, please open a new issue instead of appending to a closed one, this way it gets easily overlooked.

I don't know what you are trying to achieve, please read up on the difference between the `concat` and `merge` commands, I suspect you may be trying to use the wrong command:
http://samtools.github.io/bcftools/bcftools.html#merge
http://samtools.github.io/bcftools/bcftools.html#concat 

Regarding the order of files, that's controlled by the shell. If you type `ls *.bsg.vcf.gz`, the files will be listed in the same order as they are given to bcftools

**Comment by SallyBerry** [comment]
Hi,

I am working on assessing the genetic diversity of the RH5 Complex using VCF files. I have three compressed (.vcf.gz) files, and I need to merge them before proceeding with filtering and analysis using VCFtools.

When I attempted to merge the files, I encountered the following error:

Duplicate sample names (PF0004-C), use --force-samples to proceed anyway.

It seems that the same sample name appears in multiple VCF files, causing a conflict.

My questions are;
What is the best way to handle duplicate sample names before merging?

Should I use --force-samples, or is there a better approach (e.g., renaming samples)?

If renaming is required, how can I properly rename samples in each VCF file before merging?

I appreciate any guidance on how to resolve this issue. Thank you!
